scheduling/task_run.go

```diff
@@ -0,0 +1,160 @@
+package scheduling
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
```

Comment:
A comment at the top of the file describing what a task is would be really useful